dotcom-rendering/src/web/components/LightboxButton.importable.tsx

```diff
@@ -0,0 +1,607 @@
+			// There's no img hash so close the lightbox
+			void close();
+		} else {
+			// The reader navigated to a url that does contain an img hash. If
+			// the lightbox isn't already open, open it at that hash position.
+			if (!lightbox.hasAttribute('open')) {
+				const position = hash.substring(5);
+				void open(parseInt(position));
+			}
+		}
+	});
+}
+
+/**
+ * This overlay makes it possible to click anywhere on an image and the lightbox
+ * will hydrate/appear.
+ *
+ * How?
+ * ----
+ * ```html
+ * <gu-island deferUntil="onInteraction">
+ *   <ClickOverlay>
+ *     <button>Open lightbox</button>
+ *   </ClickOverlay>
+ * </gu-island>
+ * ```
+ *
+ * Both gu-island and ClickOverlay have click event listeners that will replay any
+ * click event to their children using synthetic events. The gu-island listener is
+ * set on page load which in turn triggers hydration for this file, adding event
+ * listeners to the ClickOverlay and button.
+ *
+ * The flow is:
+ *
+ * 1) Page loads, gu-island has a click event listener set
+ * 2) User clicks on the island (which wraps both button and overlay)
+ * 3) This file is hydrated, setting listeners on both the ClickOverlay and the button
+ * 4) gu-island fires a synthetic click event to the element originally clicked
+ * 5) If that was ClickOverlay, another synthetic event is sent to the button
+ * 6) If it was the button then the lightbox is opened
+ *
+ * Why?
+ * ----
+ * This approach means that it is possible for images to be clickable without the
+ * requirement to hydrate images - we never want to serialize images into the page! It
+ * also means we keep the feature where the lightbox code is only downloaded on
+ * interaction.
+ *
+ * What about accessibility?
+ * -------------------------
+ * The child button element is still the primary method for opening
+ * the lightbox and this is the element that we want assitive technologies like
+ * screen readers to interact with. This overlay is intentionally hidden from
+ * keyboards and screen readers and can be thought of as progressive enhancement
+ * for mouse and touch users.
+ *
+ */
+const ClickOverlay = ({ children }: { children: React.ReactNode }) => {
+	return (
+		<div
+			css={css`
+				position: absolute;
+				top: 0;
+				width: 100%;
+				height: 100%;
+				cursor: pointer;
+			`}
+			className="open-lightbox"
+		>
+			{children}
+		</div>
+	);
+};
+
+export const LightboxButton = ({
+	elementId,
+	role,
+	format,
+	isMainMedia,
+}: Props) => {
+	useEffect(() => {
+		/**
+		 * We only want to run this code once so we first check if there are any other
+		 * islands on the page that have already been hydrated (marked with gu-ready)
+		 */
+		const allIslands = document.querySelectorAll<HTMLElement>(
+			'gu-island[name="LightboxButton"]',
+		);
+		const alreadyHydrated = Array.from(allIslands).find((island) => {
+			// The island we're clicking on has already been marked as ready
+			const isIslandClickedOn =
```

Comment:
```suggestion
			const isCurrentIsland =
```

Reply:
potentially make a list of all the islands and filter out the one you are currently on (because the one you are currently on is being hydrated so it may or may not have a gu-ready label)

Reply:
After staring at this a while I realised the simpler thing to do here is to just mark the `lightbox` element itself with a `data-gu-ready` attribute and then I can just check this can not mess around with arrays of buttons

[Done here](https://github.com/guardian/dotcom-rendering/pull/7129/commits/1f168a1db84ea623badfe15fed5074bccd565448)